weldx/core.py

```diff
@@ -708,3 +711,557 @@ def units(self) -> pint.Unit:
+    """Required variable names"""
+
+    _evaluation_preprocessor: callable = None
+    """Function that should be used to adjust a var. input - (f.e. convert to Time)"""
+
+    _required_dimensions: List[str] = None
+    """Required dimensions"""
+    _required_dimension_units: Dict[str, pint.Unit] = {}
+    """Required units of a dimension"""
+    _required_dimension_coordinates: Dict[str, List] = None
+    """Required coordinates of a dimension."""
+
+    _required_unit_dimensionality: pint.Unit = None
+    """Required unit dimensionality of the evaluated expression/data"""
+
+    # do it later
+
+    _required_parameter_shape: Dict[str, int] = None
+    """Size of the parameter dimensions/coordinates - (also defines parameter order)"""
+    _alias_names: Dict[str, List[str]] = None
+    """Allowed alias names for a variable or parameter in an expression"""
+
+    # todo: other possible constraints:
+    # - allowed dimensions (not the same as allowed variables)
+
+    @classmethod
+    def _check_constraints(cls, dims, units_out: pint.Unit = None):
+        if cls._required_dimensions is not None:
+            for v in cls._required_dimensions:
+                if v not in dims:
+                    raise ValueError(f"{cls.__name__} requires dimension '{v}'.")
+        if cls._required_unit_dimensionality is not None:
+            if not units_out.dimensionality == cls._required_unit_dimensionality:
+                raise ValueError(
+                    f"{cls.__name__} requires its output unit to be of dimensionality "
+                    f"'{cls._required_unit_dimensionality}' but it actually is "
+                    f"'{units_out.dimensionality}'."
+                )
+
+    @classmethod
+    def _check_constraints_discrete(cls, data_array: xr.DataArray):
+        if cls is GenericSeries:
+            return
+
+        cls._check_constraints(data_array.dims, data_array.data.u)
+
+        if cls._required_dimension_units is not None:
+            for k, v in cls._required_dimension_units.items():
+                if k in data_array.dims and (
+                    k not in data_array.coords.keys()
+                    or U_(data_array.coords[k].attrs.get("units", "")).dimensionality
+                    != U_(v).dimensionality
+                ):
+                    raise ValueError(
+                        f"{cls.__name__} requires dimension {k} to be have the "
+                        f"unit dimensionality '{U_(v).dimensionality}'"
+                    )
+
+        if cls._required_dimension_coordinates is not None:
+            for k, v in cls._required_dimension_coordinates.items():
+                if k not in data_array.coords.keys() or not ut.compare_nested(
+                    data_array.coords[k].data.tolist(), v
+                ):
+                    raise ValueError(
+                        f"{cls.__name__} requires dimension {k} to have the "
+                        f"coordinates {v}"
+                    )
```

Comment:
if we stick to the xarray syntax here the naming is slightly confusing to me.
Dimensions cannot have values, if they do they are already promited to coordinates.

So  `_required_coordinates_values` would be a better name over `_required_dimension_coordinates`

regardless, I think we have this functionality already in `xr_check_coords` (using `values`)

Reply:
I think `_required_coordinates` is even shorter and also fits the bill. What do you think?

Reply:
also fine